src/main/java/org/openrewrite/java/testing/junit5/TimeoutRuleToClassAnnotation.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2025 the original author or authors.
+ * <p>
+ * Licensed under the Moderne Source Available License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://docs.moderne.io/licensing/moderne-source-available-license
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.testing.junit5;
+
+import org.jspecify.annotations.Nullable;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.TypeUtils;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class TimeoutRuleToClassAnnotation extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "JUnit 4 `@Rule Timeout` to JUnit Jupiter's `Timeout`";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Replace usages of JUnit 4's `@Rule Timeout` with JUnit 5 `Timeout` class annotation.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(new UsesType<>("org.junit.rules.Timeout", false), new TimeoutRuleToClassAnnotationVisitor());
+    }
+
+    public static class TimeoutRuleToClassAnnotationVisitor extends JavaIsoVisitor<ExecutionContext> {
```

Comment:
This is another outdated pattern; when there's no need for a named public reusable visitor, then it's fine to just use an anonymous visitor. Again best to apply the same wherever you copied this pattern from.